### PR TITLE
Users

### PR DESCRIPTION
**Pull request by hazmat345**
This PR moves a bunch of the user and auth stuff out of the HTTP package.

### CODE REVIEW

File: src/app/beer_garden/api/auth.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    # Convert to strings for easier comparison
+    permission_strings = set(p.value for p in required_permissions)
+
+    if not permission_strings.intersection(principal.permissions):
+        # TODO - DONT THINK SO
+        # Need to make a distinction between "you need to be authenticated
+        # to do this" and "you've been authenticated and denied"
+        # if principal == beer_garden.api.http.anonymous_principal:
+        #     raise HTTPError(status_code=401)
+        # else:
```
[the remark this answers is not in the document]
Comment (loganasherjones) [reply src/app/beer_garden/api/auth.py]: Nevermind, after reading this, this is the right place to do this check.

File: src/app/beer_garden/api/http/handlers/v1/event.py [hunk 2]
```diff
@@ -23,8 +19,6 @@ class EventSocket(AuthMixin, WebSocketHandler):
     def __init__(self, *args, **kwargs):
         super(EventSocket, self).__init__(*args, **kwargs)
 
-        self.auth_providers.append(query_token_auth)
-
```
Comment (loganasherjones) [inline src/app/beer_garden/api/http/handlers/v1/event.py]: Was there any reason this existed originally?

File: src/app/beer_garden/users.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        The updated Principal
+
+    """
+    user = db.query_unique(Principal, id=user_id)
+
+    for op in operations:
+        if op.path == "/roles":
+
+            # Updating roles always requires USER_UPDATE
+            # check_permission(self.current_user, [Permissions.USER_UPDATE])
```
Comment (loganasherjones) [inline src/app/beer_garden/users.py]: Remove this

File: src/app/beer_garden/users.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        #             )
+        #
+        #         if not custom_app_context.verify(
+        #                 current_password, self.current_user.hash
+        #         ):
+        #             raise RequestForbidden("Invalid password")
+        #
+        #     principal.hash = custom_app_context.hash(new_password)
+        #     if "changed" in principal.metadata:
+        #         principal.metadata["changed"] = True
```
Comment (loganasherjones) [inline src/app/beer_garden/users.py]: Why are we getting rid of the ability to change passwords?

**Comment by hazmat345** [comment]
This is way too old, can never be merged. Closing and deleting.

**Pull request closed without merging**